### DISCUSSION TITLE
Problem with YI Dome Guard YRS.3019 with v12 firmware

### DISCUSSION THREAD

**Discussion by akkasss** [roleoroleo/yi-hack-Allwinner-v2#671] [Q&A]
Dear friends, 

Can someone help solve such a problem? 
Some time after updating the firmware to the 12th, the camera entered a mode in which it cannot connect to the router.

 I tried to reset the camera, it doesn't help. 
The camera successfully scans the QR code, reports "Connecting to WIFI", but then again "Reading to connect". 
I read in internet that many people have such a problem after updating camera firmware to 12th.

 On SD card I found ps1.txt file (in the "no_wifi_dev" folder). 
Also, there are other files on the SD, including log-files. 

I tried to make a firmware backup, but no new files appeared on the SD.
After that, I tried to install the r35gb firmware, but apparently it does not install, because the camera behaves the same way.

Thank you so much for your help!

**Comment by roleoroleo** [comment]
You should try the unbrick procedure.

**Reply by akkasss** [reply]
However, when the camera is turned on, it turns and speaks. And i have no backups.

**Reply by roleoroleo** [reply]
I will check if I have a backup of the old version.
Are you sure your cam is s r35gb?

**Comment by akkasss** [comment]
The inscriptions on my camera look this:
https://i.ibb.co/93Z7PfV/photo-2023-07-15-14-17-41.jpg
As far as I understand, is r35gb suitable for my camera?

Is there a way to restore the old firmware version from the manufacturer, for example, the 9th?

**Reply by roleoroleo** [reply]
The label is not enough to know exactly the model.
You need the whole fw version 12.x.x.x or 9.x.x.x

**Comment by akkasss** [comment]
Maybe I can use this firmware? https://kamihome.com/firmware/

**Comment by akkasss** [comment]
How can I check the fw version if your camera refuses to connect to the router due to the fw update?

**Reply by roleoroleo** [reply]
Do you have a backup of the original fw?
I could open it and check the version.

**Reply by akkasss** [reply]
Unfortunately, I don't have a backup, because the camera stopped connecting to wi-fi unexpectedly
However, here is some kind of firmware for YI Dome Guard: https://kamihome.com/firmware/

**Comment by roleoroleo** [comment]
Please try to install r30gb hack.
If it doesn't work, I will try to send a backup firmware.

> However, here is some kind of firmware for YI Dome Guard: https://kamihome.com/firmware/

This site contains only partial firmware and you can't use them for the unbrick procedure.

**Comment by akkasss** [comment]
I seem to flash the camera (r35gb 0.2.9), since the 'Factory.done' folder appeared on the sdcard.

However, if I don't use 'configure_wi fi.cfg', then I still can't connect it to the router via YI Home using the QR code.

If I use configure_wifi.cfg, the LED starts flashing green and the camera does not talk. However, I do not find my camera in the list of devices connected to the router and I cannot connect to it with a browser, since I do not know its IP.

**Comment by akkasss** [comment]
Does YI Dome Guard YRS.3019 need to use r30gb, not r35gb firmware?

**Comment by akkasss** [comment]
Here is a backup made by your firmware: https://dropmefiles.com/iDBPN

**Reply by roleoroleo** [reply]
Sorry but I didn't have time to download the file, can you upload it again?

**Comment by akkasss** [comment]
https://dropmefiles.com/8T93X

**Comment by roleoroleo** [comment]
Yes, it's a r30gb.
Partitions are ok. So I think that a firmware restore (with v 12) won't fix your problem.
You should try to restore fw 9.

Or...
Did you try to use the new hack procedure where you can set wifi credentials?
It's not released but it's commited and I could send you a beta.

**Comment by JohnHolg** [comment]
Hi,

I installed FW 12.1.05.1 und YI-hack 0.3.2 r30gb

PTZ and rtsp-stream didn't work.

Can you help?

**Comment by xakepon** [comment]
Hello, my Dome Guard camera can't connect to Wi-Fi, please post the standard firmware and a small manual on how to flash it.

**Reply by roleoroleo** [reply]
If you want a backup partition from another user, I need to know your current version.

**Reply by xakepon** [reply]
my current version is 12.1.47.3_202410140930. I saw this version when I made a backup before flashing your software.

**Comment by roleoroleo** [comment]
Sorry but I don't have this version.
9.* or 12.1.47.2_202301121850

**Reply by xakepon** [reply]
ok. 9 or 12.1

**Reply by xakepon** [reply]
The most important thing for me is to restore the camera's functionality

**Comment by Sara-Tareq** [comment]
Hello , I'm having the same issue , I tried resetting the camera but it keeps rebooting on the connecting to wifi phase , I tried doing hack but wasn't able to install it , the model is YRS 3019 but I don't know the firmware version

**Reply by mpagdagd** [reply]
same

**Reply by roleoroleo** [reply]
I can send you a partition to restore but, if you try the wrong version, the cam will brick.

**Reply by mpagdagd** [reply]
and to avoid getting the wrong version...how do I know the right version in that case?

**Reply by roleoroleo** [reply]
Version is contained in the file /home/homever
But, to read this file you need:
- a working cam
- a backup

If you installed the hack, you should have a valid backup inside the sd card.

**Reply by hbbdgyb** [reply]
> 我可以给你发送一个分区进行恢复，但是如果你尝试了错误的版本，相机就会变砖。

Could you please share the file 9.0.05.32_202004101552 with me and let me know how to use it?

**Comment by dolagr5-eng** [comment]
Hi,
I have the same problem with my YI Dome Guard YRS 3019. The cam can't connect to wifi so i don't know the firmware version.
could you tell me how to upgrade the firmware and where i can find the firmware file? Thanks

**Reply by roleoroleo** [reply]
If you make a backup copy, you can extract the firmware version and the model suffix.

**Reply by dolagr5-eng** [reply]
Could you give me the procedure for the backup copy?

**Reply by roleoroleo** [reply]
https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card)

**Reply by dolagr5-eng** [reply]
It doesn't work. The SD still remains the same. Have you some suggestions?

**Reply by roleoroleo** [reply]
Try to change the sd card and/or reformat it with a FAT32 fs.
If you are comfortable with the soldering iron, you can connect a serial adapter and use a terminal to check the problem.

**Reply by dolagr5-eng** [reply]
that's the firmware version:
9.0.05.32_202004101552

I've another working camera. Can I extract the backup from the working camera and upload it on the not-working camera?

**Reply by roleoroleo** [reply]
Yes, if the cams are exactly the same version.

**Reply by dolagr5-eng** [reply]
ok. can you help me for the hack? 
I can connect via telnet using a wired connection.
Attempting to run the config.sh from r30gb_0.3.5.tgz results in Hack already applied, so the installer does not proceed.
The IF: if [ `grep telnetd /backup/init.sh | grep -c ^` -gt 0 ]; goes to echo "Hack already applied" 
but it isn't true.
What do you suggest?

**Reply by dolagr5-eng** [reply]
Hi, now i have a bigger issue.
I've dove the command: cat /dev/zero > /dev/mtdblock4
so I erased the main configuration and it seems that the camera doesn't read the config.sh script any more.
What can I do?

**Reply by roleoroleo** [reply]
You have to restore mtdblock4.
This partition contains the r/w file system.
You can restore it from a backup or from another cam (same model) using the unbrick procedure.

**Reply by hbbdgyb** [reply]
Could you please share the file 9.0.05.32_202004101552 with me and let me know how to use it?

**Reply by roleoroleo** [reply]
What's your goal?
Do you want to restore the original firmware or upgrade to a new version?
